backend/src/main/java/com/huellapositiva/domain/actions/UpdatePasswordAction.java

```diff
@@ -62,4 +66,28 @@ public void executePasswordChanging(String hashRecoveryPassword, String password
             throw new TimeForRecoveringPasswordExpiredException("The time to recovery password has expired");
         }
     }
+
+    /**
+     * This method updates the password in database from the profile and sends an email.
+     *
+     * @param newPassword the new password for the user
+     * @param oldPassword to check if the user has permission to change the password
+     * @param email The emails user
+     */
+    public void executeUpdatePassword(String newPassword, String oldPassword, String email) {
+
+        JpaCredential jpaCredential = jpaCredentialRepository.findByEmail(email).orElseThrow(UserNotFoundException::new);
+        PasswordHash newPasswordHash = new PasswordHash(passwordEncoder.encode(newPassword));
+
+        if (!passwordEncoder.matches(oldPassword,jpaCredential.getHashedPassword())) {
+            throw new NonMatchingPasswordException("The old password inserted does not match with the one stored in the system");
+        } else if (passwordEncoder.matches(newPassword,jpaCredential.getHashedPassword())) {
+            throw new InvalidNewPasswordException("The new password it exactly the same as the old password");
+        } else {
+            jpaCredentialRepository.updatePassword(newPasswordHash.toString(), email);
+
+            EmailAddress emailAddress = EmailAddress.from(jpaCredential.getEmail());
+            emailCommunicationService.sendConfirmationPasswordChanged(emailAddress);
+        }
```

Comment:
To improve readability move the else logic out of the if. Structure of code should follow:
`if (exceptional case) {
   throw exception
}

normal case`

Reply:
Ok, it's fixed right now.